src/bundle.rs

```diff
@@ -152,28 +152,29 @@ pub trait Authorization: fmt::Debug {
 
 /// A bundle of actions to be applied to the ledger.
 #[derive(Clone)]
-pub struct Bundle<T: Authorization, V> {
+pub struct Bundle<A: Authorization, V, D: OrchardDomain> {
     /// The list of actions that make up this bundle.
-    actions: NonEmpty<Action<T::SpendAuth>>,
+    actions: NonEmpty<Action<A::SpendAuth, D>>,
     /// Orchard-specific transaction-level flags for this bundle.
     flags: Flags,
     /// The net value moved out of the Orchard shielded pool.
     ///
     /// This is the sum of Orchard spends minus the sum of Orchard outputs.
     value_balance: V,
     /// Assets intended for burning
+    // FIXME: use BurnType like it's in Zebra? Put it as another param of Domain trait
```

Comment:
From @PaulLaux:
Can you extend?

Reply:
In `Zebra`, we introduced a special associated type `BurnType` in the `OrchardFlavour` trait:

https://github.com/QED-it/zebra/blob/tx_v6_update/zebra-chain/src/orchard/orchard_flavour.rs#L41

This type is defined as a `Vec` of burn items for the ZSA variant and as an empty type for the Vanilla variant. This makes it convenient to use in the `ShieldedData` structure, allowing the proper serialization to be automatically selected at compile time and preventing the accidental usage of a `Vec` for the Vanilla variant (this is protected and checked by the compiler).

The idea was to use the same trick in the `orchard` crate for the bundle. However, for now, we decided not to implement this, as discussed.